src/components/in-world-hud.js

```diff
@@ -0,0 +1,155 @@
+AFRAME.registerComponent("in-world-hud", {
+  schema: {
+    haptic: { type: "selector" },
+    raycaster: { type: "selector" }
+  },
+  init() {
+    this.bg = this.el.querySelector(".bg");
+    this.mic = this.el.querySelector(".mic");
+    this.nametag = this.el.querySelector(".nametag");
+    this.nametag.object3DMap.text.material.depthTest = false;
+    this.avatar = this.el.querySelector(".avatar");
+    this.data.raycaster.components.line.material.depthTest = false;
+
+    const muted = this.el.sceneEl.is("muted");
+    this.mic.setAttribute("src", muted ? "#muted" : "#unmuted");
+
+    const avatarScale = "0.1 0.1 0.1";
+    const flipXAvatarScale = "-" + avatarScale;
+
+    const scene = this.el.sceneEl;
+    this.onUsernameChanged = this.onUsernameChanged.bind(this);
+    scene.addEventListener("username-changed", this.onUsernameChanged);
+
+    this.onNametagHovered = () => {
+      this.nametag.setAttribute("color", "#00eeee");
+      this.data.haptic.emit("haptic_pulse", { intensity: "low" });
+    };
+    this.onNametagUnhovered = () => {
+      this.nametag.setAttribute("color", "white");
+    };
```

Comment:
A "down" state would be nice on these buttons.

Reply:
✔️